python/ribasim/ribasim/model.py

```diff
@@ -148,6 +149,61 @@ def _write_tables(self, directory: FilePath) -> None:
                 input_entry.write(directory, self.modelname)
         return
 
+    def validate_model(self):
+        """
+        Checks:
+        - Whether all node types in the node field are valid
+        - Whether the node IDs of the node_type fields are valid
```

Comment:
It's good to have one method `validate_model`, but let's split its subtasks into separate methods